crypto/threads_pthread.c

```diff
@@ -24,6 +24,8 @@
 #  include <unistd.h>
 #endif
 
+# include <assert.h>
+
 # ifdef PTHREAD_RWLOCK_INITIALIZER
 #  define USE_RWLOCK
 # endif
```

Comment:
Beat me to this one :)

```c
# else
#  include <assert.h>
```

is the other way of doing this.  We're only using assert() on the non _USE_RWLOCK_ paths.